### PR TITLE
Doc: pkgs{compressDrv,compressDrvWeb} improve clarity

### PR DESCRIPTION
**Pull request by hsjobeki**
## Description of changes

> Note: This doesnt get rendered into any of the manuals:
> The only place where people can discover it currently is the code and https://noogle.dev/f/pkgs/compressDrv

- Improves clarity
- More consistent with the other docs.
- Move the extensive example into an example block, which can be collapsed / is collapsed by default in nixpkgs manual.

cc: @motiejus @Ma27 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by hsjobeki** [comment]
@motiejus Somwething i missed in the initial pr:

Why do you need `...` ?
```{ formats, compressors, ... }:```

It is not possible to access the 'rest' arguments, and they are not forwarded to anything. Thus calling the function, with superflous arguments will not result in an error.

**Comment by motiejus** [comment]
This looks great, but I can only judge on the surface. Is there a simple-ish way to generate something like https://noogle.dev/f/pkgs/compressDrv locally?

> @motiejus Somwething i missed in the initial pr:
> 
> Why do you need `...` ? `{ formats, compressors, ... }:`
> 
> It is not possible to access the 'rest' arguments, and they are not forwarded to anything. Thus calling the function, with superfluous arguments will not result in an error.

This is a left-over from previous PR incarnations and can be removed. You can do it here (AFAIK), or I can do this once this PR is merged -- I let's avoid conflicts either way.

**Review by motiejus — APPROVED** [review]

**Comment by hsjobeki** [comment]
> This looks great, but I can only judge on the surface. Is there a simple-ish way to generate something like https://noogle.dev/f/pkgs/compressDrv locally to review rendered changes?
> 

No unfortunately Not. Usually you would just build the Manual. But this doesnt appear in the Manual yet.

Rendering this into the Manual is a Goal but a little complex, since we want a generic solution.


You could build noogle yourself though. This will Take a while If you do it the First time. 
You can then Override the nixpkgs-master Input with your branch. 

nix build GitHub:nix-community/noogle#ui --override-input nixpkgs-master {your url}

**Comment by motiejus** [comment]
This is perfect!

> nix build GitHub:nix-community/noogle#ui --override-input nixpkgs-master {your url}

Any flags I am missing?

```
[motiejus@vno1-gdrx:~/code/nixpkgs]$ nix build GitHub:nix-community/noogle#ui --override-input nixpkgs-master https://github.com/NixOS/nixpkgs
error: 'GitHub:nix-community/noogle#ui' is not a valid URL

[motiejus@vno1-gdrx:~/code/nixpkgs]$ 
```

**Comment by hsjobeki** [comment]
Github needs to be lowercase. Got autocorrected in my Phone 😅

**Comment by philiptaron** [comment]
I'm getting

```
error: NAR hash mismatch in input 'https://registry.npmjs.org/@corex/deepmerge/-/deepmerge-4.0.43.tgz?narHash=sha256-GlUW2/0Z1wRF1RyrT/La9POas%2Bbv9x%2BTbbCQy9V3bW8%3D', expected 'sha256-GlUW2/0Z1wRF1RyrT/La9POas+bv9x+TbbCQy9V3bW8=' but got 'sha256-Cr1OYmR1+ozY0Hwny+oQ3BHvioS5+suPrMOujPio8+w='
```

Problem on master?!

**Comment by hsjobeki** [comment]
Yep. Seems npm registry behavior changed. It'll be fixed anytime (minutes). We learn that npm sucks, and sometimes dependencies randomly change their hash ....

**Comment by motiejus** [comment]
This seems to build for this PR:

```
nix build github:nix-community/noogle#ui --override-input nixpkgs-master github:hsjobeki/nixpkgs/doc/compressDrv
```

Both http://localhost:8000/f/pkgs/compressDrvWeb and http://localhost:8000/f/pkgs/compressDrv look great. Thanks!

File: pkgs/build-support/compress-drv/default.nix [hunk 1]
```diff
@@ -4,36 +4,51 @@
...
+  «derivation /nix/store/...-spdx-license-list-data-3.24.0-compressed.drv»
   ```
 
-  See compressDrvWeb, which is a wrapper on top of compressDrv, for broader use
+  See also pkgs.compressDrvWeb, which is a wrapper on top of compressDrv, for broader usage
   examples.
+  :::
 */
 drv:
 { formats, compressors, ... }:
```
Comment (motiejus) [inline pkgs/build-support/compress-drv/default.nix:54]: perhaps

```suggestion
{ formats, compressors }:
```

while at it?

**Review by motiejus — APPROVED** [review]

File: pkgs/build-support/compress-drv/default.nix (hunk 1 above)
Comment (motiejus) [reply pkgs/build-support/compress-drv/default.nix:54]: Scratch that, there is one more place that needs, I will make a PR myself.

Comment (motiejus) [reply pkgs/build-support/compress-drv/default.nix:54]: https://github.com/NixOS/nixpkgs/pull/332558

**Review by philiptaron — APPROVED** [review]

**Pull request merged**